### DISCUSSION TITLE
I am facing an issue with the fetch function. using Nextjs, mysql, prisma

### DISCUSSION THREAD

**Discussion by assadawami22** [prisma/prisma#23393] [Q&A]
### Question

I am facing an issue with the fetch function. in the Postman using GET method  I get all the objects in the array, but when I log iot in the console I only get the first object in the array

### How to reproduce (optional)

this is the get method in the api folder 
------------------------------------------------------------------------

export async function GET(request: NextRequest) {
    const recipes = await prisma.recipe.findMany(); // Fetch all recipes from the database
    return  NextResponse.json({ data: recipes });
}
------------------------------------------------------------------------
this is fetch in the page 
------------------------------------------------------------------------

```ts
  interface recipeForm {
    id: number;
    title: string;
    ingrediants: string;
    instructions: string;
    catagory: string;
  }


  interface APIResponse {
    data: recipeForm[];
  }

  const res = await fetch( `/api/recipe`, {
    method: "GET"
  });

  const apiResponse: APIResponse = await res.json();
  const { data } = apiResponse;
  console.log(data);



  return (
    <div className="">
      {data.map(recipe => (
        <div key={recipe.id} className='grid grid-flow-row-dense grid-cols-1 gap-y-6 border rounded-md p-2'>

          <div className=' border rounded-md p-2'>
            <h1 className=" font-semibold text-gray-500 text-lg">{recipe.title}</h1>

          </div>

)

```

------------------------------------------------------------------------
here is what I get in the console
------------------------------------------------------------------------
```
[
  {
    id: 1,
    title: 'Pasta',
    ingrediants: '1 eqq, 2 flour',
    instructions: 'fiertsdc dsfwrsegf',
    catagory: 'sweet',
    createdAt: '2024-02-29T08:14:25.775Z',
    updatedAt: '2024-02-29T08:14:25.775Z'
  }
]
```
------------------------------------------------------------------------
here is what I get in PostMan
------------------------------------------------------------------------
```
{
    "data": [
        {
            "id": 1,
            "title": "Pasta",
            "ingrediants": "1 eqq, 2 flour",
            "instructions": "fiertsdc dsfwrsegf",
            "catagory": "sweet",
            "createdAt": "2024-02-29T08:14:25.775Z",
            "updatedAt": "2024-02-29T08:14:25.775Z"
        },
        {
            "id": 2,
            "title": "Pasta 111111",
            "ingrediants": "1 eqq, 2 flour",
            "instructions": "fiertsdc dsfwrsegf",
            "catagory": "sweet",
            "createdAt": "2024-03-04T10:16:30.718Z",
            "updatedAt": "2024-03-04T10:16:30.718Z"
        },
        {
            "id": 3,
            "title": "assfa",
            "ingrediants": "sfgrfewfw",
            "instructions": "wefwrftrw",
            "catagory": "sdqgvsdfg",
            "createdAt": "2024-03-04T10:17:52.570Z",
            "updatedAt": "2024-03-04T10:17:52.570Z"
        },
        {
            "id": 4,
            "title": "Pasta 11111155",
            "ingrediants": "1 eqq, 2 flour",
            "instructions": "fiertsdc dsfwrsegf",
            "catagory": "sweet",
            "createdAt": "2024-03-04T14:04:22.213Z",
            "updatedAt": "2024-03-04T14:04:22.213Z"
        },
        {
            "id": 5,
            "title": "kabsa ",
            "ingrediants": "rice, oil, dfdsx",
            "instructions": "mix rice with meat\n",
            "catagory": "lunch",
            "createdAt": "2024-03-04T14:17:35.226Z",
            "updatedAt": "2024-03-04T14:17:35.226Z"
        },
        {
            "id": 6,
            "title": ",jlkn;",
            "ingrediants": "likjn;kl.",
            "instructions": "l;kkjo",
            "catagory": "lkinjk;",
            "createdAt": "2024-03-04T14:37:03.610Z",
            "updatedAt": "2024-03-04T14:37:03.610Z"
        },
        {
            "id": 7,
            "title": "new rec 7 march",
            "ingrediants": "new rec 7 march",
            "instructions": "new rec 7 march",
            "catagory": "new rec 7 march",
            "createdAt": "2024-03-07T07:54:58.915Z",
            "updatedAt": "2024-03-07T07:54:58.915Z"
        }
    ]
}
  
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @assadawami22 👋 

Thank you for raising this question.

Can you do a `console.dir()` instead of a `console.log()`?

```js
console.dir(data, { depth: Infinity })
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
